### PR TITLE
APP-87: SDK side

### PR DESCRIPTION
**Pull request by RostakaGmfun**
Relates to https://github.com/kaaproject/sample-apps/pull/99

### CODE REVIEW

**Comment by rasendubi** [comment]
@kaa-leeroy retest this please

File: client/client-multi/client-c/toolchains/esp8266.cmake [hunk 1]
```diff
@@ -38,6 +38,10 @@ endif ()
 
 CMAKE_FORCE_C_COMPILER(${TOOLCHAIN_PATH}/bin/xtensa-lx106-elf-gcc GNU)
 
+# xtensa toolchain does not have C++ compiler, this line just suppresses detection of host C++ compiler
+CMAKE_FORCE_CXX_COMPILER(${TOOLCHAIN_PATH}/bin/xtensa-lx106-elf-gcc GNU)
```
Comment (rasendubi) [inline client/client-multi/client-c/toolchains/esp8266.cmake]: Use `${TOOLCHAIN_PATH}/bin/xtensa-lx106-elf-g++`.

**Comment by rasendubi** [comment]
@kaa-leeroy retest this please

**Comment by rasendubi** [comment]
@kaa-leeroy retest this please

**Comment by rasendubi** [comment]
Please, rebase on fresh develop.

File: client/client-multi/client-c/CMakeLists.txt [hunk 2]
```diff
@@ -191,5 +190,5 @@ if(KAA_UNITTESTS_COMPILE)
 endif()
 
 add_subdirectory(Modules/doxygen)
-
 add_subdirectory(Modules/cppcheck)
+
```
Comment (rasendubi) [inline client/client-multi/client-c/CMakeLists.txt:193]: Trailing line

File: client/client-multi/client-c/CMakeLists.txt [hunk 3]
```diff
@@ -191,5 +190,5 @@ if(KAA_UNITTESTS_COMPILE)
 endif()
 
 add_subdirectory(Modules/doxygen)
-
```
Comment (rasendubi) [inline client/client-multi/client-c/CMakeLists.txt:194]: Don't remove spacing

File: client/client-multi/client-c/CMakeLists.txt [hunk 4]
```diff
@@ -16,8 +16,7 @@
 
 cmake_minimum_required(VERSION 2.8.8)
 
-project(Kaa-c)
-enable_language(C)
+project(Kaa-c C)
```
Comment (rasendubi) [inline client/client-multi/client-c/CMakeLists.txt:19]: Please, use `project(Kaa-c LANGUAGES C)` form. It's good to be more explicit.

**Comment by rasendubi** [comment]
Minor nitpicks. Otherwise, looks good.

**Comment by forGGe** [comment]
Looks good

**Comment by rasendubi** [comment]
@RostakaGmfun please, look at Jenkins failure

**Pull request merged**